src/node_api.cc

```diff
@@ -3170,6 +3170,12 @@ napi_status napi_create_dataview(napi_env env,
   RETURN_STATUS_IF_FALSE(env, value->IsArrayBuffer(), napi_invalid_arg);
 
   v8::Local<v8::ArrayBuffer> buffer = value.As<v8::ArrayBuffer>();
+  if (byte_length + byte_offset > buffer->ByteLength()) {
+    napi_throw_range_error(env,
+                           "napi_create_dataview",
+                           "byte_offset + byte_length should be less than or "
+                           "equal to the size in bytes of the array passed in");
+  }
```

Comment:
Shouldn't there be some kind of early return in this case? Otherwise we still create the DataView instance & pass it in `*result`... right?

Reply:
You're right. I missed `return` statement. :(
Done. Thank you for review.